### PR TITLE
Add sighash types slides

### PR DESCRIPTION
**Pull request by golmit**
# Description

Add first version of the "SIGHASH Types" slide deck. The slide deck is based on the book chapter and includes examples for very simple transactions with max. 2 inputs and max. 2 outputs.

## Type of Change

- [x] Major feature (New files or assets, large changes to files)

# Checklist:

- [x] I have performed a self-review of my own changes
- [x] I have made sure that any changed LaTeX files compile properly
- [x] I have assigned at least one reviewer

### CODE REVIEW

File: assets/figures/SIGHASH_ALL.tex [hunk 1]
```diff
@@ -0,0 +1,38 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{2 inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_ALL.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_SINGLE.tex [hunk 2]
```diff
@@ -0,0 +1,45 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{2 inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_SINGLE.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_NONE.tex [hunk 3]
```diff
@@ -0,0 +1,53 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{2 inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_NONE.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_ALL_ANYONECANPAY.tex [hunk 4]
```diff
@@ -0,0 +1,47 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{x inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_ALL_ANYONECANPAY.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_NONE_ANYONECANPAY.tex [hunk 5]
```diff
@@ -0,0 +1,50 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{x inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_NONE_ANYONECANPAY.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_SINGLE_ANYONECANPAY.tex [hunk 6]
```diff
@@ -0,0 +1,42 @@
+\begin{tikzpicture}[scale=0.9, every node/.style={scale=0.9}]
+    
+        \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!15, thick, draw=black, dashed] (-4,-6.4) rectangle ++(304pt,160pt);
+        
+    \draw[color=black] plot (1.55,-0.2) node [below]
+    {\large{{Transaction}}};
+    
+    % Inputs
+    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{x inputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_SINGLE_ANYONECANPAY.tex:9]:
```suggestion
    \draw[color=black] plot (-1.35,-1.6) node[above] {\texttt{inputs:}};
```

File: assets/figures/SIGHASH_NONE_ANYONECANPAY.tex [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    % bottom left
+    \uncover<3->{
+    	\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=focus] (-3.6,-6) rectangle ++(128pt,60pt);
+    	\draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    	\draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    	\draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} \textcolor{focus}{Added by someone else}}};
+    }
+    
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{x outputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_NONE_ANYONECANPAY.tex:26]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{outputs:}};
```

File: assets/figures/SIGHASH_SINGLE_ANYONECANPAY.tex [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    % bottom left
+    \uncover<2->{
+    	\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=focus] (-3.6,-6) rectangle ++(128pt,60pt);
+    	\draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    \draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    	\draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} \textcolor{focus}{Added by someone else}}};
+    }
+    
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{x outputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_SINGLE_ANYONECANPAY.tex:26]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{outputs:}};
```

File: assets/figures/SIGHASH_ALL_ANYONECANPAY.tex [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    
+    \uncover<3>{
+    	\filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=focus] (-3.6,-6) rectangle ++(128pt,60pt);
+    	\draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    \draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    	\draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} \textcolor{focus}{Added by someone else}}};
+    }
+
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{1 output:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_ALL_ANYONECANPAY.tex:33]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{output:}};
```

File: assets/figures/SIGHASH_NONE.tex [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    \draw[color=black] plot (-3.5,-2.9)   node[right] {\small{\textbullet{} Signed}};
+    
+    % bottom left
+    \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (-3.6,-6) rectangle ++(128pt,60pt);
+    \draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    \draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    \draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} Signed}};
+
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{2 outputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_NONE.tex:24]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{outputs:}};
```

File: assets/figures/SIGHASH_SINGLE.tex [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    \draw[color=black] plot (-3.5,-2.9)   node[right] {\small{\textbullet{} Signed}};
+    
+    % bottom left
+    \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (-3.6,-6) rectangle ++(128pt,60pt);
+    \draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    \draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    \draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} Signed}};
+
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{2 outputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_SINGLE.tex:24]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{outputs:}};
```

File: assets/figures/SIGHASH_ALL.tex [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    \draw[color=black] plot (-3.5,-2.9)   node[right] {\small{\textbullet{} Signed}};
+    
+    % bottom left
+    \filldraw[yshift=-0.05cm, xshift=0.1cm,color = highlight!25, thick, draw=highlight] (-3.6,-6) rectangle ++(128pt,60pt);
+    \draw[color=black] plot (-3.5,-4.65) node[right] {\small{\textbullet{} Amount: 2 BTC}};
+%    \draw[color=black] plot (-3.5,-4.95)   node[right] {\small{\textbullet{} Anyone's input}};
+    \draw[color=black] plot (-3.5,-5.25)   node[right] {\small{\textbullet{} Signed}};
+
+	% Outputs
+    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{2 outputs:}};
```
Comment (Chainomics) [inline assets/figures/SIGHASH_ALL.tex:24]:
```suggestion
    \draw[color=black] plot (4,-1.55)   node[above] {\texttt{outputs:}};
```

File: slides/sighash_types.tex [hunk 13]
```diff
@@ -0,0 +1,107 @@
+% Choose one to switch between slides and handout
+\documentclass[]{beamer}
+%\documentclass[handout]{beamer}
+
+% Video Meta Data
+\title{Bitcoin, Blockchain and Cryptoassets}
+\subtitle{Signature Hash Type}
```
Comment (Chainomics) [inline slides/sighash_types.tex:7]:
```suggestion
\subtitle{Signature Hash Types}
```

File: slides/sighash_types.tex [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+\begin{document}
+
+\thispagestyle{empty}
+\begin{frame}[noframenumbering]
+	\titlepage
+\end{frame}
+
+\begin{frame}{Signature}
+	Network participants exchange signed transaction messages. Usually, they sign all inputs and outputs, but sometimes they may only want to protect some of them.\\
```
Comment (Chainomics) [inline slides/sighash_types.tex:24]:
```suggestion
Network participants exchange signed transaction messages.\\ \vspace{1em}
	
In most cases, they sign all in- and outputs. However, in some cases, there may be a need for a more flexible approach. As such, there is the option to only sign certain in- and outputs.\\
```

**Review by Chainomics — COMMENTED** [review]
Great job, @golmit . Some minor edits.

**Review by Chainomics — APPROVED** [review]

**Review by dariothuerkauf — COMMENTED** [review]
Great job! Didn't know much about the different SigHash Types before but this slide deck made it easy to understand in a short time. 💯

**Pull request merged**